### DISCUSSION TITLE
Nested ComboBox Menu

### DISCUSSION THREAD

**Discussion by BrianHung** [adobe/react-spectrum#6360] [Q&A]
Just saw that submenus were released! Wondering if it is possible to implement the following in react-aria:
https://ariakit.org/examples/menu-nested-combobox

Is it possible to do this with submenu and combobox, or would it require that listbox supports nesting too?

**Comment by snowystinger** [comment]
Related https://github.com/adobe/react-spectrum/discussions/5578#discussioncomment-7857680
We've built them (sub dialogs) in React Spectrum [Menu Unavailable Items](https://react-spectrum.adobe.com/react-spectrum/Menu.html#unavailable-items) from our hooks, so if you're using hooks, you can do this. We haven't built it into RAC yet.

**Reply by BrianHung** [reply]
Got it; I'll try to dig into hooks a bit more and report back if I can! Just wondering, is listbox or menu the better component here?

**Reply by snowystinger** [reply]
It depends on what you're building. If it's the same thing as in ariakit, then ListBox, I think, would be the way to go.

**Reply by snowystinger** [reply]
Submenu and subdialogs are built into RAC these days, you can see how we implemented in S2 https://github.com/adobe/react-spectrum/blob/a6e9b2504c6c5eba37823489bc36b74d144bb7c3/packages/%40react-spectrum/s2/src/Menu.tsx#L814
It's just using the SubmenuTrigger https://react-aria.adobe.com/Menu#submenus

**Comment by mwkcoding** [comment]
Did you find a solution to his? I'm trying to build a nested combobox menu

**Reply by snowystinger** [reply]
See https://github.com/adobe/react-spectrum/discussions/6360#discussioncomment-17126986
and the first example on https://react-aria.adobe.com/Autocomplete 

I don't know that I'd recommend a nested combobox menu. Combobox would be the wrong kind of pattern, it's for selecting and it sounds more like you want a searchable menu. So I'd go with the Autocomplete

**Reply by mwkcoding** [reply]
But if I want the Input to be the trigger, we cannot use Autocomplete from what I understand. It would have to be a button that would trigger the menu with an input in the popover.

**Reply by snowystinger** [reply]
That is true right now, we'll be working on the input being the trigger at some point in the near future in order to build comboboxes on top of autocomplete.

But my point was that this sounds more like a searchable menu, not a combobox. So having it be a button would probably make more sense. It can have submenus, but while searching, you should flatten the list because managing submenus with search sounds like a nightmare. You could still provide the breadcrumb path as part of each item's description so people learn where that item lives for the future.
